peer/peer.go

```diff
@@ -2109,12 +2223,22 @@ func (p *Peer) negotiateInboundProtocol() error {
 		return err
 	}
 
+	var protoVersion uint32
+	p.flagsMtx.Lock()
+	protoVersion = p.protocolVersion
+	p.flagsMtx.Unlock()
+
+	if err := p.writeSendAddrV2Msg(protoVersion); err != nil {
+		return err
+	}
```

Comment:
Since addrv2 did not come with a version bump (wtxidrelay also uses 70016 - WTXID_RELAY_VERSION in bitcoin core code), it wouldn't be possible to know if your peer supports addrv2 on the version alone (maybe they only support wtxidrelay). This approach is used by bitcoin core now to introduce new feature via messages (during handshake). I think version bumping could have been done, but maybe there's some other reason. In other words, I dont know

Reply:
Yeh agree that a version bump should have been done instead, as the current interactive negotiation stuff just serves to make the logic to figure out which features you can use more complicated, vs the way we do things on the LN side re looking at the feature bit intersection.